### PR TITLE
Create and upgrade multiple k3s clusters with validation tests

### PR DESCRIPTION
**Pull request by rancher-max**
- Adds Jenkins job to import multiple k3s clusters, run preupgrade, upgrade, and run postupgrade jobs
- Adds upgrade_cluster code for k3s clusters
- Removes some extraneous prints and "multinode" text from k3s creation code

### CODE REVIEW

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_k3s_upgrade [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+              }
+
+              sh "./tests/v3_api/scripts/configure.sh"
+              sh "./tests/v3_api/scripts/build.sh"
+            }
+
+            stage('Deploy Rancher Server') {
+              try {
+                if (!env.CATTLE_TEST_URL || !env.ADMIN_TOKEN || !env.USER_TOKEN ||
+                (env.CATTLE_TEST_URL == "" && env.ADMIN_TOKEN == "" && env.USER_TOKEN == "")) {
```
Comment (ShylajaDevadiga) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_k3s_upgrade:64]: Can we simply this condition? Maybe using trim

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_k3s_upgrade [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            stage('Deploy k3s Clusters') {
+              try {
+                  echo "Deploying k3s clusters. Versions: ${RANCHER_K3S_VERSIONS}. Names: ${RANCHER_CLUSTER_NAMES}"
+                  jobs = [:]
+                  k3s_versions = RANCHER_K3S_VERSIONS.split(",")
+                  cluster_arr = RANCHER_CLUSTER_NAMES.split(",")
+                  cluster_count = cluster_arr.size()
+                  RANCHER_UPGRADE_CHECK = "preupgrade"
+                  RANCHER_VALIDATE_RESOURCES_PREFIX = "mystep1"
+                  RANCHER_CREATE_RESOURCES_PREFIX = "mystep1"
```
Comment (ShylajaDevadiga) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_k3s_upgrade:96]: Upgrade variables are not used in deploy cluster.

File: tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_k3s_upgrade [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+            }
+
+            stage('Run Postupgrade Tests in Parallel') {
+              try {
+                  jobs = [:]
+                  cluster_arr = env.RANCHER_CLUSTER_NAMES.split(",")
+                  cluster_count = cluster_arr.size()
+                  RANCHER_UPGRADE_CHECK = "postupgrade"
+                  RANCHER_VALIDATE_RESOURCES_PREFIX = "mystep1"
+                  RANCHER_CREATE_RESOURCES_PREFIX = "mystep2"
```
Comment (ShylajaDevadiga) [inline tests/validation/tests/v3_api/scripts/pipeline/Jenkinsfile_k3s_upgrade:193]: Is it possible to reuse the above code as it is just the PREFIX change?

**Review by ShylajaDevadiga — DISMISSED** [review]
LGTM

**Review by sowmyav27 — CHANGES_REQUESTED** [review]

File: tests/validation/tests/v3_api/test_custom_host_reg.py [hunk 4]
```diff
@@ -141,6 +141,11 @@ def test_deploy_rancher_server():
             i += 1
         validate_cluster_state(client, cluster)
         env_details += "env.CLUSTER_NAME='" + cluster.name + "'\n"
+
+    if UPDATE_KDM is True:
```
[the remark this answers is not in the document]
Comment (rancher-max) [reply tests/validation/tests/v3_api/test_custom_host_reg.py]: I put it before now! Thank you!

**Review by sowmyav27 — APPROVED** [review]
LGTM

**Review by bmdepesa — APPROVED** [review]
LGTM

**Pull request merged**